### PR TITLE
Make changes to allow for mac users

### PR DESCRIPTION
**Pull request by werne2j**
- Add instructions for Mac users
- Change scripts to allow for a variable OS (linux or mac)

**Best attempt at some of the logic, it doesn't feel great so if you have a better way something could be done, i will gladly make the change

### CODE REVIEW

File: playbooks/roles/os_temps/tasks/check_new_app.yml [hunk 1]
```diff
@@ -25,12 +25,12 @@
 
 # Setup tag if current container image build is successful
 - name: Modify tags on images
-  shell: "{{ oc_bin }} get imagestream | awk '{print $2}' | grep -v DOCKER | sed 's/.*5000\\///g' | grep '{{ build_config_name_files[template_name] }}' | xargs -i {{ oc_bin }} tag {}:latest {}:{{ tag }}"
+  shell: "{{ oc_bin }} get imagestream | awk '{print $2}' | grep -v DOCKER | sed 's/.*5000\\///g' | grep '{{ build_config_name_files[template_name] }}' | xargs -I{} {{ oc_bin }} tag {}:latest {}:{{ tag }}"
```
Comment (arilivigni) [inline playbooks/roles/os_temps/tasks/check_new_app.yml]: Do we need to change this to -I vs -i?

File: playbooks/roles/os_temps/tasks/setup_containers.yml [hunk 2]
```diff
@@ -40,15 +40,15 @@
   ignore_errors: yes
 
 - name: "Cleanup all failed dc, bc, routes, svc, and imagestreams on the cluster"
-  shell: "{{ oc_bin }} get all | grep '{{ failed_container_name }}' | awk '{print $1}' | egrep -v 'builds\\/|po\\/' | xargs -i {{ oc_bin }} delete {}"
+  shell: "{{ oc_bin }} get all | grep '{{ failed_container_name }}' | awk '{print $1}' | egrep -v 'builds\\/|po\\/' | xargs -I{} {{ oc_bin }} delete {}"
```
Comment (arilivigni) [inline playbooks/roles/os_temps/tasks/setup_containers.yml]: Do we need to change this to -I vs -i?

File: playbooks/roles/os_temps/tasks/setup_containers.yml [hunk 3]
```diff
@@ -40,15 +40,15 @@
...
   with_items: "{{ oc_check_app_status.stdout_lines }}"
   loop_control:
     loop_var: failed_container_name
   ignore_errors: yes
   when: oc_check_app_status.stdout != ""
 
 - name: "Cleanup any serviceaccounts, pvc, and rolebindings for an app if it exists on the cluster"
-  shell: "{{ oc_bin }} get {{ item }} | egrep '{{ oc_check_app_status.stdout_lines|join('|') }}' | awk '{print $1}' | xargs -i {{ oc_bin }} delete {{ item }}/{}"
+  shell: "{{ oc_bin }} get {{ item }} | egrep '{{ oc_check_app_status.stdout_lines|join('|') }}' | awk '{print $1}' | xargs -I{} {{ oc_bin }} delete {{ item }}/{}"
```
Comment (arilivigni) [inline playbooks/roles/os_temps/tasks/setup_containers.yml]: Do we need to change this to -I vs -i?

File: playbooks/roles/os_temps/tasks/check_new_app.yml [hunk 4]
```diff
@@ -25,12 +25,12 @@
...
   when: modify_tags|bool == true and current_build_succeeded|bool == true
 
 # Output the OpenShift build logs if the current build failed
 - name: Output the OpenShift build logs if the current build failed
-  shell: "echo \"{{ oc_build_result.stdout }}\" | awk '{print $1}' | xargs -i {{ oc_bin }} logs builds/{}"
+  shell: "echo \"{{ oc_build_result.stdout }}\" | awk '{print $1}' | xargs -I{} {{ oc_bin }} logs builds/{}"
```
Comment (arilivigni) [inline playbooks/roles/os_temps/tasks/check_new_app.yml]: Do we need to change this to -I vs -i?

File: playbooks/roles/os_temps/tasks/check_new_app.yml (hunk 1 above)
Comment (werne2j) [reply playbooks/roles/os_temps/tasks/check_new_app.yml]: Oh whoops, i meant to leave a comment on this. `-i` is deprecated and `-I` is the preferred option. Also, `-i` does not exist for the bsd xargs command.

http://man7.org/linux/man-pages/man1/xargs.1.html
```
-i[replace-str], --replace[=replace-str]
              This option is a synonym for -Ireplace-str if replace-str is
              specified.  If the replace-str argument is missing, the effect
              is the same as -I{}.  This option is deprecated; use -I
              instead.
```

Comment (arilivigni) [reply playbooks/roles/os_temps/tasks/check_new_app.yml]: Does this address all instances of xargs -i?

Comment (werne2j) [reply playbooks/roles/os_temps/tasks/check_new_app.yml]: I went through and changed all of them from `-i` to `-I{}`, as according to the man page they have the effect

**Pull request merged**